src/components/molecules/Visualizer/Layers/store.ts

```diff
@@ -68,6 +73,72 @@ export class LayerStore {
     return l;
   }
 
+  #usedIds: string[] = [];
+  #newId = () => {
+    const genResId = () =>
+      "_xxxxxxxxxxxxxxxxxxxxxxxxx".replace(/[x]/g, function () {
+        return ((Math.random() * 16) | 0).toString(16);
+      });
+    let id;
+    do {
+      id = genResId();
+    } while (this.#usedIds.includes(id));
+    this.#usedIds.push(id);
+    return id;
+  };
+
+  #insertLayer = (layerData: AddedLayerData, save = false) => {
+    const tar = layerData.parentId ? this.#map.get(layerData.parentId) : this.#root;
+    if (!tar?.children) return;
+    tar.children.push(layerData.layer);
+    const flattenedLayers = flattenLayers([layerData.layer]);
+    flattenedLayers.forEach(l => {
+      this.#map.set(l.id, l);
+      this.#pmap.set(l.id, this.#pluginLayer(l));
+    });
+    save && !tar.creator && this.#addedLayersData.push(layerData);
```

Comment:
```suggestion
    if (save && !tar.creator) this.#addedLayersData.push(layerData);
```

- IMO (I like if statement in normal codes, except JSX)
- Tell me why `!tar.creator` is needed